apps/api/src/app/auth/usecases/password-reset-request/password-reset-request.usecase.ts

```diff
@@ -55,6 +57,16 @@ export class PasswordResetRequest {
     };
   }
 
+  private static getResetRedirectLink(token: string, src?: PasswordResetFlowEnum): string {
+    switch (src) {
+      case PasswordResetFlowEnum.USER_PROFILE:
+        return `${process.env.FRONT_BASE_URL}/settings/profile?token=${token}`;
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches and the blocking or non-blocking decorations to further communicate your intent.  -->
**suggestion:** I think we should add a step of validation here to check only the users with no password available can access this API.
Currently, anyone can access it.‏